release/rancher/rancher.go

```diff
@@ -296,3 +335,144 @@ func createPRFromRancher(ctx context.Context, rancherBaseBranch, title, branchNa
+		if err != nil {
+			return err
+		}
+
+		scanner := bufio.NewScanner(strings.NewReader(string(repoContent)))
+		lineNum := 1
+
+		for scanner.Scan() {
+			lineByte, line := formatLineByte(scanner.Text())
+
+			if devDependencyPattern.Match(lineByte) {
+				badFiles = true
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				lineContentLower := strings.ToLower(lineContent.Content)
+				if strings.Contains(lineContentLower, "chart") {
+					content.ChartsWithDev = append(content.ChartsWithDev, lineContent)
+				}
+				if strings.Contains(lineContentLower, "kdm") {
+					content.KDMWithDev = append(content.KDMWithDev, lineContent)
+				}
+			}
+			if strings.Contains(filePath, "/package/Dockerfile") {
+				if !strings.Contains(line, "_VERSION") {
+					continue
+				}
+				matches := regexp.MustCompile(`CATTLE_(\S+)_MIN_VERSION`).FindStringSubmatch(line)
+				if len(matches) == 2 && strings.Contains(line, "-rc") {
+					lineContent := ContentLine{Line: lineNum, File: filePath, Content: formatContentLine(line)}
+					content.MinFilesWithRC = append(content.MinFilesWithRC, lineContent)
+				}
+			}
+			if rcTagPattern.Match(lineByte) {
+				badFiles = true
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				content.FilesWithRC = append(content.FilesWithRC, lineContent)
+			}
+			lineNum++
+		}
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+	}
+
+	tmpl := template.New("rancher-release-rc-dev-deps")
+	tmpl = template.Must(tmpl.Parse(templateCheckRCDevDeps))
+	buff := bytes.NewBuffer(nil)
+	err = tmpl.ExecuteTemplate(buff, "componentsFile", content)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(buff.String())
+
+	if forCi && matchCommitMessage && badFiles {
+		return errors.New("check failed, some files don't match the expected dependencies for a final release candidate")
+	}
+
+	return nil
+}
+
+func writeRancherImagesDeps(content *Content, rcTagPattern *regexp.Regexp) error {
+	// files below were generated in build time into rancher path
+	imageFiles := []string{"./bin/rancher-images.txt", "./bin/rancher-windows-images.txt"}
+
+	for _, file := range imageFiles {
+		filePath, err := os.Open(file)
+		if err != nil {
+			continue
+		}
+		defer filePath.Close()
+
+		scanner := bufio.NewScanner(filePath)
+		lineNumber := 1
+		for scanner.Scan() {
+			lineByte, line := formatLineByte(scanner.Text())
+
+			if rcTagPattern.Match(lineByte) {
+				lineContent := ContentLine{Line: lineNumber, File: file, Content: formatContentLine(line)}
+				content.RancherImages = append(content.RancherImages, lineContent)
+			}
+			lineNumber++
+		}
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func formatContentLine(line string) string {
+	re := regexp.MustCompile(`\s+`)
+	line = re.ReplaceAllString(line, " ")
+	return strings.TrimSpace(line)
+}
+
+func formatLineByte(line string) ([]byte, string) {
```

Comment:
I don't think the TrimSpace is necessary either, since the return value is only used to match a regular expression where the space does not matter, and as the input to formatContentLine where it will be trimmed again.

Reply:
Removed